CMakeLists.txt

```diff
@@ -95,3 +95,74 @@ function(trusted_firmware_build)
   endif()
 
 endfunction()
+
+# Gets a list of commands to be performed after a successful Zephyr build that
+# involves trusted-firmware-m.
+function(trusted_firmware_get_post_build_commands out_commands)
```

Comment:
Not sure exactly where / which target will call this function, but I think it would be better to have this as a macro which actually adds this as a real post build step, like:
```
macro(sign_trusted_firmware target <other args>)
  if(NOT CONFIG_TFM_BL2_FALSE)
    <existing code>

    add_custom_command(TARGET ${target}
                         POST_BUILD
                          COMMAND ${PYTHON_EXECUTABLE} ${TFM_MCUBOOT_DIR}/scripts/imgtool.py
                          sign
                          <existing additional args>
    )

   <and so on>
  endif()
endmacro()
```

Then users won't need to handle a `out_commands` variable and decide what to do with it.

The reason to suggest a macro here, instead of a function, has to do with the `add_custom_commnad(... POST_BUILD ..)` must be called in same CMakeLists directory as the target.

Also, feel free to find a better name.
Not sure if: `sign_trusted_firmware` is the best name.

If this is purely a Zephyr macro, then maybe it should be called `zephyr_<something>`.

Reply:
> Not sure exactly where / which target will call this function, but I think it would be better to have this as a macro which actually adds this as a real post build step, like:

I should of course look here:
https://github.com/zephyrproject-rtos/zephyr/pull/28806/files

and that actually means this will be hard to change, due to the nature of:
`extra_post_build_commands`

If changing that, then we are looking into a completely different task.